tests/test_oml/test_functional/test_metrics/test_retrieval_metrics.py

```diff
@@ -165,15 +170,160 @@ def compare_metrics(
+
+    metrics_expected["cmc"][1] = torch.tensor([1, 1, 0, 0, 1, 0]).float()
+    metrics_expected["precision"][1] = torch.tensor([1, 1, 0, 0, 1, 0]).float()
+    metrics_expected["map"][1] = torch.tensor([1, 1, 0, 0, 1, 0]).float()
+
+    metrics_expected["cmc"][2] = torch.tensor([1, 1, 1, 0, 1, 0]).float()
+    metrics_expected["precision"][2] = torch.tensor([1, 0.5, 0.5, 0, 1, 0]).float()
+    metrics_expected["map"][2] = torch.tensor([1, 0.5, 0.25, 0, 1, 0]).float()
+
+    metrics_expected["cmc"][3] = torch.tensor([1, 1, 1, 1, 1, 0]).float()
+    metrics_expected["precision"][3] = torch.tensor([0.6666, 0.3333, 0.6666, 0.3333, 1, 0]).float()
+    metrics_expected["map"][3] = torch.tensor([0.6666, 0.3333, 0.3888, 0.1111, 1, 0]).float()
+
+    metrics_expected["cmc"][4] = torch.tensor([1, 1, 1, 1, 1, 1]).float()
+    metrics_expected["precision"][4] = torch.tensor([1, 0.6666, 0.6666, 0.6666, 1, 1]).float()
+    metrics_expected["map"][4] = torch.tensor([0.9166, 0.5, 0.3888, 0.2777, 1, 0.25]).float()
+
+    metrics_expected["cmc"][5] = torch.tensor([1, 1, 1, 1, 1, 1]).float()
+    metrics_expected["precision"][5] = torch.tensor([1, 1, 1, 1, 1, 1]).float()
+    metrics_expected["map"][5] = torch.tensor([0.9166, 0.7, 0.5888, 0.4777, 1, 0.25]).float()
+
+    metrics_expected["cmc"][10] = torch.tensor([1, 1, 1, 1, 1, 1]).float()
+    metrics_expected["precision"][10] = metrics_expected["precision"][5]
+    metrics_expected["map"][10] = metrics_expected["map"][5]
+
+    return mask_gt, gt_tops, top_k, metrics_expected
+
+
+def test_calc_cmc(exact_test_case: Tuple[torch.Tensor, torch.Tensor, Tuple[int, ...], TMetricsDict]) -> None:
+    mask_gt, gt_tops, top_k, metrics_expected = exact_test_case
+    cmc = calc_cmc(gt_tops, top_k)
+    for k, cmc_val in zip(top_k, cmc):
+        assert torch.all(
+            torch.isclose(cmc_val, metrics_expected["cmc"][k], atol=1.0e-4)
+        ), f"cmc@{k} expected: {metrics_expected['cmc'][k]}; evaluated: {cmc_val}."
+
+
+@pytest.mark.parametrize("top_k", (1, 2, 3, 4, 5, 10))
+def test_calc_cmc_individual(
+    exact_test_case: Tuple[torch.Tensor, torch.Tensor, Tuple[int, ...], TMetricsDict], top_k: int
+) -> None:
+    mask_gt, gt_tops, _, metrics_expected = exact_test_case
+    cmc = calc_cmc(gt_tops, (top_k,))[0]
+    assert torch.all(
+        torch.isclose(cmc, metrics_expected["cmc"][top_k])
+    ), f"cmc@{top_k} expected: {metrics_expected['cmc'][top_k]}; evaluated: {cmc}."
+
+
+@pytest.mark.parametrize("top_k", (tuple(), (0, -1), (0,), (1.5, 2), (1.0, 2.0)))
+def test_calc_cmc_check_params(top_k: Tuple[int, ...]) -> None:
+    with pytest.raises(ValueError):
+        gt_tops = torch.ones((10, 5), dtype=torch.bool)
+        calc_cmc(gt_tops, top_k)
+
+
+def test_calc_precision(exact_test_case: Tuple[torch.Tensor, torch.Tensor, Tuple[int, ...], TMetricsDict]) -> None:
+    mask_gt, gt_tops, top_k, metrics_expected = exact_test_case
+    n_gt = _calc_n_gt(mask_gt)
```

Comment:
if you will decide to move `_calc_n_gt` calculation inside the function, you can also use

```python
@pytest.mark.parametrize("metric_function, metric_name", [(calc_cmc, 'cmc'), (calc_precision, 'precision'), (calc_map, 'map')])
``` 

and use as assert
```python
assert torch.all(
            torch.isclose(metrics_calculated, metrics_expected[metric_name][k], atol=1.0e-4)
        ), f"{metric_name}@{k} expected: {metrics_expected[metric_name][k]}; evaluated: {metrics_calculated}."
```